lib/class-wp-json-comments-controller.php

```diff
@@ -0,0 +1,470 @@
+<?php
+
+/**
+ * Access comments
+ */
+class WP_JSON_Comments_Controller extends WP_JSON_Controller {
+
+	/**
+	 * Get a list of comments.
+	 *
+	 * @param  WP_JSON_Request $request Full details about the request.
+	 * @return WP_Error|WP_JSON_Response|mixed
+	 */
+	public function get_items( $request ) {
+		$prepared_args = $this->prepare_items_query( $request );
+
+		$query = new WP_Comment_Query;
+		$comments = $query->query( $prepared_args );
+
+		foreach ( $comments as &$comment ) {
+			$post = get_post( $comment->comment_post_ID );
+			if ( ! $this->check_read_post_permission( $post ) || ! $this->check_read_permission( $comment ) ) {
```

Comment:
Rather than scattering a bunch of duplicate checks in the controller, can we have `check_read_permission()` also check `check_read_post_permission()` if the comment is assigned to a post?